perun-core/src/main/java/cz/metacentrum/perun/core/api/GroupsManager.java

```diff
@@ -902,4 +902,17 @@
 	 * @throws PrivilegeException
 	 */
 	List<Group> getGroupUnions(PerunSession sess, Group group, boolean reverseDirection) throws InternalErrorException, GroupNotExistsException, PrivilegeException;
+
+	/**
+	 * Move one group structure under another group in same vo
+	 *
+	 * @param sess perun session
+	 * @param destinationGroup group to which is moving group moved
+	 * @param movingGroup group which is moved to destination group
+	 *
+	 * @throws InternalErrorException
+	 * @throws GroupNotExistsException
+	 * @throws PrivilegeException
+	 */
+	void moveGroup(PerunSession sess, Group destinationGroup, Group movingGroup) throws InternalErrorException, GroupNotExistsException, PrivilegeException, GroupMoveNotAllowedException;
```

Comment:
We should support the case "make group top-level", in such case ``destinationGroup`` would be null. Please allow it in a method implementation and update corresponding check before throwing ``GroupMoveNotAllowedException``.